### ISSUE TITLE
QP problems give strange results when [ is preceded by coefficient or minus

### ISSUE THREAD

**Issue by fuglede** [ERGO-Code/HiGHS#846]
Consider the following LP file:

```
Minimize
 obj: [ - x1^2 ]/2
Bounds
 0 <= x1 <= 40
End
```

When running this through HiGHS, we get an error:

```
Running HiGHS 1.2.2 [date: 2022-05-04, git hash: d1d9839b-dirty]
Copyright (c) 2022 ERGO-Code under MIT licence terms
ERROR:   Hessian has 1 diagonal entries in [-1, 0) so is not positive semidefinite
Error loading file
```

This is expected and makes sense. But now, let's move the negative sign to the front:
```
Minimize
 obj: - [ x1^2 ]/2
Bounds
 0 <= x1 <= 40
End
```

This time we don't get an error, even though the objective is the same one:

```
Running HiGHS 1.2.2 [date: 2022-05-04, git hash: d1d9839b-dirty]
Copyright (c) 2022 ERGO-Code under MIT licence terms
QP   qp3 has 0 rows; 1 cols; 0 matrix nonzeros; 1 Hessian nonzeros
0, -1.000000, 0, 0.000973, 0.000000, 0, 0.000000, 0.000000
1, -1.000000, 0, 0.001478, 0.000000, 0, 0.000000, 0.000000
Model   status      : Optimal
QP ASM    iterations: 1
Objective value     : -1.0000000000e+00
HiGHS run time      :          0.00
```

According to the solution, the value of `x1` for this optimal result is 0, and the objective is somehow -1. I'm guessing the input is parsed as `-1 + [ x1^2 ]`.

Following this hunch, let's try the following:

```
Minimize
 obj: 3 [ x1^2 ]/2
Bounds
 0 <= x1 <= 40
End
```

Here we get this:

```
Running HiGHS 1.2.2 [date: 2022-05-04, git hash: d1d9839b-dirty]
Copyright (c) 2022 ERGO-Code under MIT licence terms
QP   qp3 has 0 rows; 1 cols; 0 matrix nonzeros; 1 Hessian nonzeros
0, 3.000000, 0, 0.001115, 0.000000, 0, 0.000000, 0.000000
1, 3.000000, 0, 0.001485, 0.000000, 0, 0.000000, 0.000000
Model   status      : Optimal
QP ASM    iterations: 1
Objective value     :  3.0000000000e+00
HiGHS run time      :          0.00
```

Again, the optimal value of `x1` is 0.

**Comment by fuglede** [comment]
Here's another weird one. Not sure if it's related; looks a bit different:

```
$ cat qp3.lp
Maximize
 obj: x2 + [ - x1^2 ]/2
Bounds
 0 <= x1 <= 40
End

$ highs qp3.lp --solution_file sol
Running HiGHS 1.2.2 [date: 2022-05-04, git hash: d1d9839b-dirty]
Copyright (c) 2022 ERGO-Code under MIT licence terms
QP   qp3 has 0 rows; 2 cols; 0 matrix nonzeros; 2 Hessian nonzeros
0, 0.000000, 0, 0.001376, 0.000000, 0, 0.000000, 0.000000
4, -nan, 1, 0.001940, 0.000000, 0, 0.000000, 1.000000
Model   status      : Optimal
QP ASM    iterations: 4
Objective value     :              -nan
HiGHS run time      :          0.00

$ cat sol
Model status
Optimal

# Primal solution values
Feasible
Objective 0
# Columns 2
x2 0
x1 0
# Rows 0

# Dual solution values
Feasible
# Columns 2
x2 0
x1 0
# Rows 0

# Basis
HiGHS v1
None
```

**Comment by odow** [comment]
I think part of the problem is that the LP file format is ill-defined and too flexible. Do other solvers handle complexity like this? Otherwise, it's reasonable to make the producers of LP files tidy the terms before passing to HiGHS.

**Comment by feldmeier** [comment]
Thanks for this, I'll see if this can be fixed easily or if its more involved

**Comment by fuglede** [comment]
> Do other solvers handle complexity like this? 

I tried with Xpress:

* It fails as expected on `Minimize [ - x1^2 ]/2` (?899 Warning: The quadratic objective is not convex)
* It also fails as expected on `Minimize - [ x1^2 ]/2` (?899 Warning: The quadratic objective is not convex)
* It chokes on `Minimize 3 [ x1^2 ]/2` (?506 Error: A variable name was expected at (2.9):)

> Otherwise, it's reasonable to make the producers of LP files tidy the terms before passing to HiGHS.

Absolutely, and I would agree that the `3 [ x1^2 ] / 2` example is pushing what one can expect to work, but `- [ x1^2 ]/2` being silently translated to `- 1 + [ x1^2 ] / 2` isn't ideal; in both cases, a parser error would be better.

**Comment by jajhall** [comment]
Closed by #873
